src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.Editor.cs

```diff
@@ -370,8 +387,13 @@ private async Task<(Document, bool addedFields)> GenerateFieldDelegatingConstruc
                                 }
                                 else
                                 {
-                                    // Can change the parameter name, so do so.
-                                    var newParameterName = new ParameterName(newFieldName, isFixed: false);
+                                    // Can change the parameter name, so do so.  
+                                    // But first remove any prefix added due to field naming styles
+                                    var fieldNameMinusPrefix = newFieldName.Substring(fieldNamingRule.NamingStyle.Prefix.Length);
+                                    var proposedParameterName = parameterNamingRule.SymbolSpecification != null
+                                        ? parameterNamingRule.NamingStyle.MakeCompliant(fieldNameMinusPrefix).First()
+                                        : fieldNameMinusPrefix;
```

Comment:
`parameterNamingRule.SymbolSpecification != null`  is a hack to see if we have a naming rule or not.  Is there a better way?

Reply:
I think it would be easier to just add a parameter naming rule to `FallbackNamingRules.RefactoringMatchLookupRules`. You could add the symbol to the existing camelCase field naming rule in that collection (but not the one with the prefix).